src/v/iceberg/tests/remove_snapshots_action_test.cc

```diff
@@ -81,6 +81,31 @@ add_tag(snapshot_id id, const ss::sstring& tag_name, table_metadata* table) {
     return table->refs->at(tag_name);
 }
 
+// Examines the updates from the transaction and checks that they match the
+// expected counts, also ensuring that snapshot removals are stored
+// individually, to match the expectations from Java implementations.
+// https://github.com/apache/iceberg/blob/3e6da2e5437ffb3f643275927e5580cb9620256b/core/src/main/java/org/apache/iceberg/MetadataUpdateParser.java#L550-L553
+void check_expected_updates(
+  const transaction& txn,
+  size_t expected_snap_removals,
+  size_t expected_ref_removals) {
+    size_t snapshot_removals = 0;
+    size_t ref_removals = 0;
+    for (const auto& u : txn.updates().updates) {
+        if (std::holds_alternative<remove_snapshots>(u)) {
+            auto& r = std::get<remove_snapshots>(u);
+            EXPECT_EQ(1, r.snapshot_ids.size());
+            snapshot_removals++;
+        } else if (std::holds_alternative<remove_snapshot_ref>(u)) {
+            ref_removals++;
```

Comment:
Makes sense to me that this restriction wouldn't apply to refs, but are we certain about that?

Reply:
Yea, the remove_snapshot_ref update doesn't have a list at all -- it only contains the reference name to remove

Reply:
Ah, gotcha